tailormap-components/projects/core/src/lib/user-interface/panel-resizer/layout.service.ts

```diff
@@ -0,0 +1,248 @@
+/**============================================================================
+ * Manages the positioning and sizing of components on the page.
+ *
+ * Remarks: For now, very simple layout manager!!!!!!
+ *          For now, only Dock.Bottom components are supported!!!!!!!
+ *          For now, only one Dock.Bottom component is tested!!!!!!
+ *
+ * @@HostListener in Service - KAN NIET
+ * @@https://stackoverflow.com/questions/39592972/is-it-possible-to-use-hostlistener-in-a-service-or-how-to-use-dom-events-in-an
+ *===========================================================================*/
+
+import { Injectable, ElementRef, OnDestroy } from '@angular/core';
+import { Subject, fromEvent } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import { PanelResizerComponent} from './panel-resizer.component';
+
+/**----------------------------------------------------------------------------
+ * Defines the mandatory properties of a panel managed by this layout service.
+ */
+export interface ILayoutComponent {
+  canResize: boolean;
+  componentRef: ElementRef;
+  dock: Dock;
+  initialHeight: number;
+  panelResizer: PanelResizerComponent;
+  windowState: WindowState;
+  getCaptionbarHeight(): number;
+}
+
+/**----------------------------------------------------------------------------
+ */
+export enum Dock {
+  Top,
+  Bottom,
+  Fill
+}
+
+/**----------------------------------------------------------------------------
+ */
+export enum WindowState {
+  Maximized,
+  Minimized,
+  Initial,
+  Closed = 3
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LayoutService implements OnDestroy {
```

Comment:
Is dit specifiek voor de attributelist, of te gebruiken voor andere componenten? Indien dat laatste, dan zou ik hem 1 niveau hoger zetten: dan is het duidelijk wat de relatie is tov andere componenten

Reply:
Gedaan.